modules/indexer/code/indexer.go

```diff
@@ -137,37 +136,46 @@ func Init() {
 	// Create the Queue
 	switch setting.Indexer.RepoType {
 	case "bleve", "elasticsearch":
-		handler := func(data ...queue.Data) []queue.Data {
+		handler := func(items ...*IndexerData) []*IndexerData {
 			idx, err := indexer.get()
 			if idx == nil || err != nil {
 				log.Error("Codes indexer handler: unable to get indexer!")
-				return data
+				return items
+			}
+			if !idx.Ping() {
```

Comment:
So we need to `ping` every time even if it will run normally?

Reply:
`Ping` is quite cheap (comparing to the `index`) below.

And we have "batch", so there won't be frequent ping for every item.


If you still consider it is a problem, we can move the `Ping` into the loop, only handle the failed index requests.

Reply:
I think move the `Ping` into the loop is better, because the es may be stopped in the middle of one batch operation. So we will lose some of the tasks if we check before the loop.

Reply:
done in 0f65bbb1f2a10dd3ea114a953ed7c2d9085f5ce9